lib/checkWebHook.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+var bufferEq = require('buffer-equal-constant-time');
+var Cesium = require('cesium');
+var crypto = require('crypto');
+
+var defined = Cesium.defined;
+
+var Settings = require('./Settings');
+module.exports = checkWebHook;
+
+function checkWebHook(req, res, next) { // eslint-disable-line no-unused-vars
+    // check header fields
+    var id = req.headers['x-github-delivery'];
+    if (!defined(id)) {
+        next(new Error('No id found in the request'));
+        return;
+    }
+
+    var event = req.headers['x-github-event'];
+    if (!defined(event)) {
+        next(new Error('No event found in the request'));
+        return;
+    }
+
+    var sign = req.headers['x-hub-signature'] || '';
+    if (Settings.secret && !sign) {
+        next(new Error('No signature found in the request'));
+        return;
+    }
+
+    if (!defined(req.body)) {
+        next(new Error('Make sure body-parser is used'));
```

Comment:
This error message is misleading.  If a post body is not defined, it's not because we're no using body-parser, it's because whomever issued the POST didn't send one.  I would just put something like "Expected req.body to be defined").